Remove duplicate destination decoding

promag opened the pull request:
Follow up of #11117, this patch removes an extra unnecessary destination decoding that was identified while reviewing #11117. It is also the only case where `IsValidDestinationString` is called before `DecodeDestination`.

For reference see [comment](https://github.com/bitcoin/bitcoin/pull/11117#discussion_r137145517).

promag committed:
Remove duplicate destination decoding

laanwj commented:
utACK https://github.com/bitcoin/bitcoin/pull/11259/commits/6bc6e241176b4a3acd1be2c12b7ffa7f1f65bd56

sipa commented:
utACK 6bc6e241176b4a3acd1be2c12b7ffa7f1f65bd56

MeshCollider commented:
utACK https://github.com/bitcoin/bitcoin/commit/6bc6e241176b4a3acd1be2c12b7ffa7f1f65bd56

promag committed:
Remove unused GetKeyID and IsScript methods from CBitcoinAddress

promag commented:
As per @laanwj suggestion on IRC, added a commit to remove unused methods that caused
```
base58.cpp:308:6: warning: ‘bool {anonymous}::CBitcoinAddress::IsScript() const’ defined but not used [-Wunused-function]
base58.cpp:298:6: warning: ‘bool {anonymous}::CBitcoinAddress::GetKeyID(CKeyID&) const’ defined but not used [-Wunused-function]
```
Also rebased.

sipa commented:
utACK 8d0041e607a0e5bff52152f3d8af504f3703baa7

laanwj commented:
Thanks. re-utACK 86e6dd4b6, builds without warnings (except in leveldb, which we can't do anything about here) now.
